src/modules/vibeCheck.module.ts

```diff
@@ -18,27 +21,73 @@ async function handleVibeCheckMessage(message: Message | PartialMessage) {
     }
 
     // check lockout
-    const vibeLock = await getLastVibed(message.author.id)
-    if (!vibeLock) {
-        updateLastVibed(message.author.id)
+    const lastVibeResponse = await getLastNVibes(message.author.id, 1)
+    if (!lastVibeResponse || lastVibeResponse.length === 0) {
         doVibeCheck(message)
     } else {
-        const timeDiff = Date.now() - vibeLock.lastvibed
+        const lastVibe = lastVibeResponse[0]
+        const timeDiff = Date.now() - lastVibe.time
         if (timeDiff > LOCKOUT_MILLIS) {
-            updateLastVibed(message.author.id)
             doVibeCheck(message)
         } else {
-            const minutes = Math.floor((LOCKOUT_MILLIS - timeDiff) / MILLIS_IN_SECOND / SECONDS_IN_MINUTE)
-            message.channel.send(`You must wait ${minutes} minute(s) before doing another vibe check.`)
+            const minutes = Math.ceil((LOCKOUT_MILLIS - timeDiff) / MILLIS_IN_SECOND / SECONDS_IN_MINUTE)
+            const noun = minutes > 1 ? "minutes" : "minute"
+            message.channel.send(`You must wait ${minutes} ${noun} before doing another vibe check.`)
         }
     }
 }
 
+async function handleLastNVibesMessage(message: Message | PartialMessage) {
+    if (!message.author || !message.channel) {
+        return
+    }
+
+    const match = message.content!.match(/\d+/)
+    if (!match) {
+        console.log("Something terrible has happened")
+        return
+    }
+
+    const n = parseInt(match[0])
+    if (n > LAST_N_VIBES_MAX) {
+        await message.channel.send(`You may only see your last ${LAST_N_VIBES_MAX} vibes.`)
+        return
+    }
+
+    const lastNVibesResponse = await getLastNVibes(message.author.id, parseInt(match[0]))
+
+    if (!lastNVibesResponse || lastNVibesResponse.length === 0) {
+        await message.channel.send("You have no previous vibes.")
+        return
+    }
+
+    const lastNVibes: number[] = lastNVibesResponse.map(v => v.vibe)
+    const trueN: number = lastNVibes.length
+    const sortedVibes: number[] = [...lastNVibes].sort((a, b) => a - b)
+
+    const sum = sortedVibes.reduce((prev, cur) => prev + cur)
+    const avg = sum / trueN
+    const median = trueN % 2 == 0 ?
+         (sortedVibes[Math.floor(trueN / 2)] + sortedVibes[Math.ceil(trueN / 2)]) / 2 :
+         sortedVibes[Math.floor(trueN / 2)]
+
+    await message.channel.send(`**Your last ${trueN} vibes:**\n` +
+        lastNVibes.join(", ") + "\n\n" +
+        `**Average:** ${avg.toFixed(1)}\n` +
+        `**Median:** ${median.toFixed(1)}\n`)
+}
+
 export const VibeCheckModule: BotModule = (client: Client) => {
     console.log("Loaded VibeCheckModule")
     client.on("message", async message => {
-        if (message.channel && message.content === "vibe check") {
+        if (message.channel && message.content === "vibbe check") {
```

Comment:
I think this should be changed back to vibe check

Reply:
done